seacrowd/sea_datasets/tydiqa/tydiqa.py

```diff
@@ -0,0 +1,469 @@
+                                "id": str(id_),
+                                "question_id": title,
+                                "document_id": title,
+                                "question": question,
+                                "type": "multiple_choice",
+                                "choices": [""],
+                                "context": doc,
+                                "answer": [""],
+                                # the schema of 'meta' aren't specified either to allow some flexibility
+                                "meta": {
+                                    "passage_answer_candidates": {
+                                        "plaintext_start_byte": start_byte,
+                                        "plaintext_end_byte": end_byte,
+                                    },
+                                    "annotations": {
+                                        "passage_answer_candidate_index": passage_cand_answers,
+                                        "minimal_answers_start_byte": min_answers_start_byte,
+                                        "minimal_answers_end_byte": min_answers_end_byte,
+                                        "yes_no_answer": yes_no_answers,
+                                    },
+                                    "language": lang,
+                                },
+                            }
+                    elif self.config.name == "primary_task_thai_seacrowd_qa":
+                        if lang == "thai":
+                            yield id_, {
+                                "id": str(id_),
+                                "question_id": title,
+                                "document_id": title,
+                                "question": question,
+                                "type": "multiple_choice",
+                                "choices": [""],
+                                "context": doc,
+                                "answer": [""],
+                                # the schema of 'meta' aren't specified either to allow some flexibility
+                                "meta": {
+                                    "passage_answer_candidates": {
+                                        "plaintext_start_byte": start_byte,
+                                        "plaintext_end_byte": end_byte,
+                                    },
+                                    "annotations": {
+                                        "passage_answer_candidate_index": passage_cand_answers,
+                                        "minimal_answers_start_byte": min_answers_start_byte,
+                                        "minimal_answers_end_byte": min_answers_end_byte,
+                                        "yes_no_answer": yes_no_answers,
+                                    },
+                                    "language": lang,
+                                },
+                            }
+                    else:
```

Comment:
Could you implement some helper functions reduce the verbosity of transforming the raw data to the schematized data? Looks like we can reuse the implementation for each group (source and seacrowd_qa)

Reply:
thanks, add helper function